### PR TITLE
Add imageTagTransformer

### PR DESCRIPTION
**Pull request by Liujingfang1**
part of #161
created imageTagTransformer, the `Transform` function will do
- loop through the resources
- For every resource, find its `containers` sessions
- For every `image` in `containers`, match the image `name` with the image names in the `ImageTags` list
- If found a match, replace the image `name` by `ImageTag.Name:ImageTag:NewTag`

### CODE REVIEW

File: pkg/transformers/imagetag.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// replaceTag replaces the image tags inside one object
+// It loops though all images inside containers session, finds matched ones and update the tag name
+func (pt *imageTagTransformer) replaceTag(obj map[string]interface{}) error {
```
Comment (monopole) [inline pkg/transformers/imagetag.go]: please break this long method into shorter ones.  too much indentation

Comment (Liujingfang1) [reply pkg/transformers/imagetag.go]: I split it into two methods. See if it is clear and easy to understand.

File: pkg/transformers/imagetag.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		if !found {
+			continue
+		}
+		for _, imagetag := range pt.imageTags {
+			if isImageMatched(image.(string), imagetag.Name) {
+				container["image"] = strings.Join([]string{imagetag.Name, imagetag.NewTag}, ":")
+				break
+			}
+		}
+		containers[i] = container
```
Comment (monopole) [inline pkg/transformers/imagetag.go]: Does the test pass without this assignment?

File: pkg/transformers/imagetag.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+	return nil
+}
+
+func isImageMatched(s, t string) bool {
+	imagetag := strings.Split(s, ":")
+	if len(imagetag) >= 1 {
+		return imagetag[0] == t
+	}
+	return false
```
Comment (monopole) [inline pkg/transformers/imagetag.go]: just 
```
return len(imagetag) >= 1 && imagetag[0] == t
```

File: pkg/transformers/imagetag.go (hunk 2 above)
Comment (Liujingfang1) [reply pkg/transformers/imagetag.go]: Just tested it and it passed. I'll remove this assignment

**Pull request merged**